Multiple Project updates

matrei opened the pull request:
Please see individual commit messages for detailed descriptions of each change.

Breaking change:
`MongoSpec` in `org.grails.plugins:mongodb` has been moved to `testFixtures`.

matrei committed:
fix: enable correct Spring auto-configuration in `boot-plugin`

The Spring auto-configuration was incorrectly set in the `boot-plugin`, causing misconfiguration issues. This commit ensures the correct auto-configuration is applied.

matrei committed:
ci: fix typo in `renovate.json` for `grails-datastore-async`

Corrected a typo in the package name grails-datastore-async in renovate.json to ensure proper dependency updates.

matrei committed:
ci: update GitHub workflows

Update the GitHub workflows to the latest Grails repository standard and best practices.

matrei committed:
test: fix `MongoDbGormAutoConfigurationSpec` failure due to misconfiguration

Fixed a Spring misconfiguration that caused MongoDbGormAutoConfigurationSpec to fail. This commit ensures the test runs correctly.

matrei committed:
build: extract common build logic into `*-config.gradle` files

Moved shared build configuration logic into `*-config.gradle` files, enabling a more compositional approach to build setup.

matrei committed:
test: update `BsonPersistentEntityCodecSpec` for updated API

Refactored `BsonPersistentEntityCodecSpec` to align with the updated API. The constructor `JsonWriterSettings(JsonMode)`` no longer exists, requiring adjustments to the test.

matrei committed:
test: refactor `BookSpec` to not use try/catch in setup

matrei committed:
test: remove Micronaut HttpClient dependency from `GlobalTemplateSpe`

Refactored GlobalTemplateSpec to eliminate the use of Micronaut HttpClient, removing the project's dependency on it.

matrei committed:
test: re-enable two passing tests in `MongoDbDataStoreSpringInitializerSpec`

Removed the `@Ignore` annotation from two tests in `MongoDbDataStoreSpringInitializerSpec` since they are now passing, allowing them to run as part of the test suite.

matrei committed:
build: update develocity build scan and remote cache

matrei committed:
test: clean up config in test applications

matrei committed:
refactor: remove `grails-app` from Grails MongoDB Plugin

Remove unused `grails-app` directory.

The `application.yml` setting `grails.gorm.failOnError: false` seems to be a
left-over from copy-pasting the `gorm-hibernate` plugin.

matrei committed:
fix: update `MongodbGrailsPlugin` for Grails 7

matrei committed:
chore: cleanup unused imports

matrei committed:
test: replace `@Ignore` with `@PendingFeature` in `LocalMongoUnitSpec`

Update a test in `LocalMongoUnitSpec` to use `@PendingFeature` instead of `@Ignore`,
allowing it to run while marking it as an expected failure until the issue is resolved.

matrei committed:
test: replace deprecated `Package.getPackage()` usage in `MongoSpec`

Refactor `MongoSpec` to remove the deprecated `Package.getPackage()` call,
ensuring compatibility with newer Java versions.

matrei committed:
test: cleanup `UrlMappings` in `examples/gson-templates`

It looks like this file has been copy/pasted from another project.

matrei committed:
build: refactor build configuration and update dependencies

- Adopt a composition-based build structure.
- Integrate the Grails Publish Plugin.
- Update dependencies and refine dependency scopes.
- Add support for the `excludeUnusedTransDeps` flag to allow building and testing with unused transitive dependencies excluded.

matrei committed:
ci: update mongodb settings

- Update 7.0.14 to 7.0 which uses the latest 7.0.x version
- Add mongodb server to groovy-joint-workflow.yml

jdaugherty commented on .github/workflows/release.yml:
```diff
@@ -1,87 +1,81 @@
+        id: release_version
+        run: echo "value=${GITHUB_REF:11}" >> $GITHUB_OUTPUT
+      - name: "⚙️ Run pre-release"
+        uses: grails/github-actions/pre-release@15e3017e73c5c4bfdc02c36792da171418da563f # 2024-12-22
```
Since we own that github action, do we really need to pin it?

matrei replied:
Well, it pointed to the main branch, which felt kind of scary.

jdaugherty replied:
I think we would only update this if we wanted to update all of them. I'm ok tagging it if you want to fix it that way too.

matrei replied:
Yes, I think a non-moving target, like a commit sha or tag, or at least a major version i better.

jdaugherty replied:
I tagged the latest to v3.0 https://github.com/grails/github-actions/releases/tag/v3.0

jdaugherty commented on .github/workflows/release.yml:
```diff
@@ -1,87 +1,81 @@
-      - name: Determine docs target repository
+        run: ./gradlew docs:docs
+      - name: "🛫 Export Gradle Properties"
+        uses: grails/github-actions/export-gradle-properties@15e3017e73c5c4bfdc02c36792da171418da563f # 2024-12-22
```
Since we own that github action, do we really need to pin it?

matrei replied:
See above

jdaugherty commented on .github/workflows/gradle.yml:
```diff
@@ -13,79 +13,84 @@ on:
+          if_false: ${{ github.repository }}
+      - name: "📤 Publish to Github Pages"
+        if: steps.docs.outcome == 'success'
+        uses: grails/github-pages-deploy-action@f1bddf4d126e4375529c0f397b028eb5ac9a13e6 # 2024-12-22
```
Since we own that github action, do we really need to pin it?

matrei replied:
See above

jdaugherty commented on .github/workflows/release.yml:
```diff
@@ -1,87 +1,81 @@
+      - name: "📤 Publish to Github Pages"
         if: success()
-        uses: grails/github-pages-deploy-action@main
+        uses: grails/github-pages-deploy-action@f1bddf4d126e4375529c0f397b028eb5ac9a13e6 # 2024-12-22
```
Since we own that github action, do we really need to pin it?

matrei replied:
See above

jdaugherty commented on .github/workflows/release.yml:
```diff
@@ -1,87 +1,81 @@
-          token: ${{ secrets.GITHUB_TOKEN }}
-        env:
-          SNAPSHOT_SUFFIX: -SNAPSHOT
+        uses: grails/github-actions/post-release@15e3017e73c5c4bfdc02c36792da171418da563f # 2024-12-22
```
Since we own that github action, do we really need to pin it?

matrei replied:
See above

jdaugherty commented on gradle.properties:
```diff
@@ -1,19 +1,23 @@
+
+asciidoctorGradleVersion=4.0.4
+datastoreVersion=9.0.0-SNAPSHOT
 grailsViewsVersion=4.0.0-M1
```
I think this can be removed now

matrei committed:
ci: update mongodb versions

- Switch to only use major version.
- Add version 8.

jdaugherty commented on examples/grails3-hibernate5-mongodb/build.gradle:
```diff
@@ -1,43 +1,39 @@
 
-group "examples"
+apply plugin: 'groovy'
+apply plugin: 'org.grails.grails-web'
```
There are gsp files in this project so why not apply both web + gsp?

matrei replied:
To my knowledge, the gsp gradle plugin is only used when packaging the app. We are only running bootRun (or bootTestRun).

jdaugherty replied:
Ah, that makes sense.  What are your thoughts about having the gsp plugin extend the web plugin?  I'm trying to find ways to simplify gradle configuration for grails apps for easier adoption.

matrei replied:
Since there are discussions about decoupling GSP from Grails, this approach might not align with that direction.

jdaugherty replied:
Good point.  Let's follow-up on this discussion later.  I'm taking a closer look at the gradle plugin after the amount of problems it's caused us.

jdaugherty commented on examples/grails3-mongodb-database-per-tenant/build.gradle:
```diff
@@ -1,37 +1,30 @@
+group = 'examples'
+
+apply plugin: 'groovy'
+apply plugin: 'org.grails.grails-web'
```
There are gsps in this project so why aren't both web + gsp applied?

matrei replied:
See above

jdaugherty commented on examples/grails3-mongodb/build.gradle:
```diff
@@ -1,44 +1,34 @@
 
-group "examples"
+apply plugin: 'groovy'
+apply plugin: 'org.grails.grails-web'
```
There are gsps in this project so why aren't both the web & gsp plugins applied?

matrei replied:
See above

jdaugherty commented on examples/grails3-mongodb/build.gradle:
```diff
@@ -1,44 +1,34 @@
-plugins {
-    id "com.bertramlabs.asset-pipeline" version "5.0.4"
```
I assume we removed the asset pipeline gradle plugin since we never package this app?

matrei replied:
Yes

jdaugherty commented on examples/gson-templates/build.gradle:
```diff
@@ -1,51 +1,31 @@
 
-group = 'functional.tests'
+apply plugin: 'groovy'
+apply plugin: 'org.grails.grails-web'
```
This project has gsps, why aren't we using that plugin as well?

matrei replied:
See above

jdaugherty commented on examples/gson-templates/grails-app/controllers/functional/tests/UrlMappings.groovy:
```diff
@@ -11,24 +11,6 @@ class UrlMappings {
-        "401"(controller: 'error', action: 'unauthorized')
         "404"(view:'/notFound')
-
-        "/books"(resources:"book")
```
I don't think we can remove all of these.  The rest based actions won't be defined by the default mapping above and are still needed.  We should leave these so the test app can be used for debugging, etc.

matrei replied:
But there is no book resource in this application.

jdaugherty replied:
That's what I missed!  Thank you.

jdaugherty commented on examples/test-data-service/build.gradle:
```diff
@@ -1,52 +1,33 @@
 
-apply plugin: "java"
+apply plugin: 'groovy'
+apply plugin: 'org.grails.grails-web'
```
Isn't the views plugin needed to compile the gson views?  gsp can be left off since there aren't any gsps.

matrei replied:
The views plugin is not needed to run bootRun, bootTestRun.

jdaugherty replied:
Thank you.

jdaugherty commented on gradle/publish-config.gradle:
```diff
@@ -0,0 +1,24 @@
+import org.grails.gradle.plugin.publishing.GrailsPublishExtension
+
+if (project.hasProperty('mavenSnapshotPublishUrl')) {
```
What do you think about just hard coding the publish url in the gradle files and not even bothering to pass it in gradle?  Seems like dual maintenance to put it in 2 locations.

matrei replied:
I was contemplating on this. It is the name `mavenPublishUrl` that does not feel right in the project gradle files. This is why I added this, so you understand directly by looking at the name what it is.

jdaugherty replied:
the grails publish plugin was written to switch between the maven & central repos.  the url is ignored based on whether it's a release or not.  Would it feel better if in the publish section you added these values: 

     grailsPublish {
          snapshotPublishType = PublishType.MAVEN_PUBLISH
          releasePublishType = PublishType.NEXUS_PUBLISH
     }

matrei replied:
No, this makes me even more confused 😄.

I think it is confusing, because to me, they are both publishing to Maven.
The difference is which repo it publishes to. It's just that one is publishing to a Nexus staging repo.
I understand the difficulties involved here in making this easy as I reviewed the feature.
I just think, if a new developer reads these build files, this will make it clear what we are actually setting.

If you don't like it, I can remove it.

jdaugherty replied:
Let's leave it.  It's just good to talk through these.  Thank you.

jdaugherty commented on grails-plugin/src/main/groovy/grails/plugins/mongodb/MongodbGrailsPlugin.groovy:
```diff
@@ -15,22 +15,19 @@ import org.springframework.core.env.PropertyResolver
-    def grailsVersion = "6.0.0 > *"
+    def license = 'Apache 2.0 License'
+    def organization = [name: 'Grails', url: 'https://grails.org/']
+    def developers = [[name: 'Puneet Behl', email: 'behlp@objectcomputing.com']]
```
We've been removing email addresses for any distributed class.

jdaugherty commented on grails-plugin/src/main/groovy/grails/plugins/mongodb/MongodbGrailsPlugin.groovy:
```diff
@@ -15,22 +15,19 @@ import org.springframework.core.env.PropertyResolver
-    def title = "GORM MongoDB"
+    def loadAfter = ['domainClass', 'hibernate5', 'hibernate6', 'services']
+    def author = 'Puneet Behl'
+    def authorEmail = 'behlp@objectcomputing.com'
```
We've been removing email addresses for any distributed class

jdaugherty commented on json-templates/build.gradle:
```diff
@@ -1,22 +1,25 @@
-ext.set('projectDesc', 'JSON Views Templates')
+
+ext {
+    projectDesc = 'JSON Views Templates'
```
* The developers will mismatch on this library compared to previous published.  Before it was just Puneet.
* The title will mismatch compared to previously published (but I think that's ok)

matrei replied:
Good catch!

jdaugherty replied:
FYI: When  I did my merge, I had changed the description to: 'Provides JSON Views for MongoDB to the Grails framework.'  Since these views are only used for mongo.

jdaugherty reviewed (changes requested):
Great work!  I'll merge the docs updates & remove the rx gorm docs after we get this merged.

matrei committed:
chore: remove author and developer info from `MongodbGrailsPlugin`

Author and developer metadata have been removed from MongodbGrailsPlugin, as this information is not necessary for core plugins.

matrei committed:
build: fix maven pom info for `views-json-templates`

matrei committed:
chore: remove unused version property `grailsViewsVersion`

matrei committed:
ci: use v3 tag for `grails/github-actions`

matrei committed:
ci: use v3 tag for `grails/github-pages-deploy-action`

jdaugherty reviewed (approved):
It looks like you also removed the author/developers reference.  We've only been removing the email addresses, but I'm ok with removing these too.  They're in the published artifact anyhow.
